Redefined scale of traffic lights and stop sign, modified rotation to…

masskro0 opened the pull request:
… be intuitive, bug fix of pole textures, added examples for stop signs and traffic lights

masskro0 committed:
Redefined scale of traffic lights and stop sign, modified rotation to be intuitive, bug fix of pole textures, added examples for stop signs and traffic lights

TrackerSB commented on examples/trafficLightStopSignTest/trafficLightsStopSignTest.dbc.xml:
```diff
@@ -0,0 +1,38 @@
+                <camera width="600" height="400" fov="120" direction="FRONT" id="egoFrontCamera" />
+            </ai>
+            <movement>
+                <waypoint x="1.0" y="0.0" tolerance="2" movementMode="AUTONOMOUS" />
```
Is this duplication of the initial state position required? If so, I should fix that. Otherwise please remove this line to make clear to everyone looking at this example that it is not required.

masskro0 replied:
Do you mean the first waypoint and the initialState tag?

TrackerSB replied:
Yes. The first waypoint contains the same information as the initial state.

TrackerSB commented on examples/trafficLightStopSignTest/trafficLightsStopSignTest.dbc.xml:
```diff
@@ -0,0 +1,38 @@
+            <not>
+                <scSpeed participant="ego" limit="10" />
+            </not>
+        </vcPosition>
```
VCs! Very nice!

TrackerSB commented on simnode/generator.py:
```diff
@@ -156,44 +156,47 @@ def add_obstacles_to_scenario(self, scenario: Scenario) -> None:
                 mesh = ProceduralBump(pos, rot, obstacle.width, obstacle.length, height, obstacle.upper_length,
                                       obstacle.upper_width)
             elif obstacle_type is Stopsign:
+                rot = (0, 0, -(obstacle.z_rot - 90))
```
This assumes that the environment that DriveBuild uses is flat.
Just in theory: At some day in the future it may allow hills or pits. As soon as this is possible one wants to define xRot and yRot as well.
Another nailcare: "-(obstacle.z_rot - 90)" is equivalent to "90 - obstacle.z_rot" and I think it is easier to read.

TrackerSB commented on simnode/generator.py:
```diff
@@ -156,44 +156,47 @@ def add_obstacles_to_scenario(self, scenario: Scenario) -> None:
                 name_pole = "pole" + str(id_number)
-                traffic_light = StaticObject(name=name_light, pos=(obstacle.x, obstacle.y, 5.32), rot=rot,
+                rot=(0, 0, -(obstacle.z_rot) - 90)
+                traffic_light = StaticObject(name=name_light, pos=(obstacle.x, obstacle.y, 4.62), rot=rot,
```
Here you could consider hills and pits easily as well. So please consider obstacle.z as well.

TrackerSB commented on simnode/generator.py:
```diff
@@ -156,44 +156,47 @@ def add_obstacles_to_scenario(self, scenario: Scenario) -> None:
                 name_light2 = "trafficlight" + str(id_number) + 'a'
                 name_pole = "pole" + str(id_number)
-                rad = radians(obstacle.z_rot)
+                rad = radians(rot[2])
```
As soon as traffic signs allow to be rotated with xRot and yRot as well, this calculation potentially changes.

masskro0 committed:
Added rotation of all axis, improved example of traffic lights and stop sign placement, removed height as a required property

TrackerSB commented on examples/trafficLightStopSignTest/trafficLightsStopSignTest.dbe.xml:
```diff
@@ -0,0 +1,25 @@
+        <trafficlightdouble x="30" y="-3"/>
+        <trafficlightdouble x="80" y="26" zRot="45"/>
+
+        <!-- Just to show that you can also rotate these objects on every axis -->
```
Or to simulate broken signs. 👍

TrackerSB commented on simnode/dbtypes/scheme.py:
```diff
@@ -51,7 +51,6 @@ class InitialState:
 class Obstacle(ABC):
     x: float
     y: float
-    height: float
```
👍

TrackerSB commented on simnode/dbtypes/scheme.py:
```diff
@@ -51,7 +51,6 @@ class InitialState:
-    height: float
     # oid: Optional[str] = None  # NOTE Limitation of inheritance from dataclasses
     # x_rot: Optional[float] = 0  # NOTE Limitation of inheritance from dataclasses
     # y_rot: Optional[float] = 0  # NOTE Limitation of inheritance from dataclasses
```
For sake of striving for perfection you could add another comment similar to these for "z" such that it makes clear that every obstacle is supposed to have a z value.

Commit:
Introduce default z coordinate

The generator is able to handle z values as well. Since the environment is currently flat the default z value is set to a default of zero.

Commit:
Mention limitations of dataclasses

TrackerSB committed:
Merge branch 'master' into master
